apps/api/src/app/environments/usecases/get-my-environments/get-my-environments.usecase.ts

```diff
@@ -1,26 +1,46 @@
-import { Injectable, Logger, Scope } from '@nestjs/common';
+import { Injectable, Logger, NotFoundException, Scope } from '@nestjs/common';
+
 import { EnvironmentEntity, EnvironmentRepository } from '@novu/dal';
+import { decryptApiKey } from '@novu/application-generic';
+
 import { GetMyEnvironmentsCommand } from './get-my-environments.command';
+import { EnvironmentResponseDto } from '../../dtos/environment-response.dto';
 
 @Injectable({
   scope: Scope.REQUEST,
 })
 export class GetMyEnvironments {
   constructor(private environmentRepository: EnvironmentRepository) {}
 
-  async execute(command: GetMyEnvironmentsCommand): Promise<EnvironmentEntity[]> {
+  async execute(command: GetMyEnvironmentsCommand): Promise<EnvironmentResponseDto[]> {
     Logger.verbose('Getting Environments');
 
     const environments = await this.environmentRepository.findOrganizationEnvironments(command.organizationId);
 
+    if (!environments?.length)
+      throw new NotFoundException(`Environments for organization ${command.organizationId} not found`);
+
     return environments.map((environment) => {
       if (environment._id === command.environmentId) {
-        return environment;
+        return this.decryptApiKeys(environment);
       }
 
       environment.apiKeys = [];
 
       return environment;
     });
   }
+
+  private decryptApiKeys(environment: EnvironmentEntity): EnvironmentResponseDto {
+    const res = { ...environment };
```

Comment:
I think it's always better to be descriptive with naming, because there's always a chance the function could be extended in the future.
```suggestion
    const decryptedApiKeysEnvironment = { ...environment };
```

Reply:
Agree, those types of variable names created are usually temporary while developing, and then I forget to update them in the end.
Thanks for highlighting i am always for naming improvement.